src/hathor.c

```
@@ -15,6 +15,11 @@ void sha256d(uint8_t *in, size_t inlen, uint8_t *out) {
     cx_sha256_t hash;
     uint8_t buffer[32];
 
+    // Check for NULL pointers
+    if ((in == NULL) || (out == NULL)) {
+        return;
```

Are failing silently here?